src/util.js

```diff
@@ -645,24 +645,25 @@ function mapDoc(doc, callback) {
 /**
  * split text into whitespaces and words
  * @param {string} text
- * @param {{ splitCjkText: boolean }} options
  * @return {Array<{ type: "whitespace", value: " " | "" } | { type: "word", value: string }>}
  */
-function splitText(text, options) {
+function splitText(text) {
   const KIND_NON_CJK = "non-cjk";
   const KIND_CJK_CHARACTER = "cjk-character";
   const KIND_CJK_PUNCTUATION = "cjk-punctuation";
 
   const nodes = [];
 
-  (!options.splitCjkText
-    ? text
-    : text.replace(
-        new RegExp(`(${cjkRegex.source})\n(${cjkRegex.source})`, "g"),
-        "$1$2"
-      )
-  )
-    .split(/(\s+)/g)
+  text
+    .replace(
+      new RegExp(`(${cjkRegex.source})\n(${cjkRegex.source})`, "g"),
+      "$1$2"
+    )
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp
+    // `\s` but exclude full-width whitspace (`\u3000`)
+    .split(
+      /([ \f\n\r\t\v\u00a0\u1680\u2000-\u200a\u2028\u2029\u202f\u205f\ufeff]+)/g
```

Comment:
This regex can be written as `/([^\S\u3000]+)/`. Not sure if that's better. (It can even be written as `/((?:(?!\u3000)\s)+)/`, but that's definitely not better!)

(Also note that the `g` flag is not needed for `.split()`.)